XiEditor/StyleMap.swift

```diff
@@ -203,6 +203,23 @@ class StyleMapState: UnfairLock {
                   underline: $0.underline, italic: $0.italic, weight: $0.weight)
         } }
     }
+
+    func measureWidth(id: Int, s: String) -> Double {
+        let builder = TextLineBuilder(s, font: self.font)
+        let range = NSMakeRange(0, s.utf16.count)
+        applyStyle(builder: builder, id: id, range: range)
+        return builder.measure()
+    }
+
+    func measureWidths(_ args: [[String: AnyObject]]) -> [[Double]] {
```

Comment:
I'd keep in mind whether in this particular case it would be worth trying to stay in `NSDictionary` land.